mlx/coverity_services.py

```diff
@@ -1,253 +1,203 @@
-DEFECT_STATUS_ATTR_NAME = "DefectStatus"
-CLASSIFICATION_ATTR_NAME = "Classification"
-ACTION_ATTR_NAME = "Action"
-COMMENT_ATTR_NAME = "Comment"
-
-
-def parse_two_part_term(term, delim=','):
-    '''Parse a term assuming [ [part1],[part2] ]'''
-    valid = False
-    part1 = ""
-    part2 = ""
-    if term.find(delim) != -1:
-        valid = True
-        field1 = term.split(delim, 1)[0]
-        if bool(field1):
-            part1 = field1
-        field2 = term.rsplit(delim, 1)[-1]
-        if bool(field2):
-            part2 = field2
-    return valid, part1, part2
-
-
-def compare_strings(str_a, str_b):
-    '''Compare strings for equivalence
-
-    some leniency allowed such as spaces and casing
-    '''
-    if re.match(str_b, str_a, flags=re.IGNORECASE):
-        return True
-    # ignore embedded spaces and some odd punctuation characters ("todo" = "To-Do")
-    str_a2 = re.sub(r'[.:\-_ ]', '', str_a)
-    str_b2 = re.sub(r'[:\-_ ]', '', str_b)  # don't remove dot (part of regex?)
-    if re.match(str_b2, str_a2, flags=re.IGNORECASE):
-        return True
-    return False
-
-
-class Service:
-    '''
-    Basic endpoint Service
-    '''
-
-    def __init__(self, transport, hostname, port, ws_version=DEFAULT_WS_VERSION):
-        self.set_transport(transport)
-        self.set_hostname(hostname)
-        self.set_port(port)
-        self.set_ws_version(ws_version)
-        self.client = None
-
-    def set_transport(self, transport):
-        '''Set transport protocol'''
-        self.transport = transport
-
-    def get_transport(self):
-        '''Get transport protocol'''
-        return self.transport
-
-    def set_hostname(self, hostname):
-        '''Set hostname for service'''
-        self.hostname = hostname
-
-    def get_hostname(self):
-        '''Get hostname for service'''
-        return self.hostname
-
-    def set_port(self, port):
-        '''Set port for service'''
-        self.port = port
-
-    def get_port(self):
-        '''Get port for service'''
-        return self.port
-
-    def set_ws_version(self, ws_version):
-        '''Set WS version for service'''
-        self.ws_version = ws_version
-
-    def get_ws_version(self):
-        '''Get WS version for service'''
-        return self.ws_version
-
-    def get_service_url(self, path='', add_port=True):
-        '''Get Service url with given path'''
-        url = self.transport + '://' + self.hostname
-        if self.port and add_port:
-            url += ':' + self.port
-        if path:
-            url += path
-        return url
-
-    def get_ws_url(self, service):
-        '''Get WS url with given service'''
-        return self.get_service_url('/ws/' + self.ws_version + '/' + service + '?wsdl')
+ACTION_LIST = {
+    "Undecided",
+    "Fix Required",
+    "Fix Submitted",
+    "Modeling Required",
+    "Ignore",
+    "On hold",
+    "For Interest Only",
+}
+
+
+class CoverityDefectService:
+    """
+    Coverity Defect Service (WebServices)
```

Comment:
```suggestion
    Convenience class for retrieving data from the Coverity REST API
```
A Web Service is different from an API.